bitcoin/src/util/taproot.rs

```diff
@@ -282,12 +282,9 @@ impl TaprootSpendInfo {
         for leaves in node.leaves {
             let key = (leaves.script, leaves.ver);
             let value = leaves.merkle_branch;
-            match info.script_map.get_mut(&key) {
-                Some(set) => {
-                    set.insert(value);
-                    continue; // NLL fix
-                }
-                None => {}
+            if let Some(set) = info.script_map.get_mut(&key) {
+                set.insert(value);
+                continue; // NLL fix
             }
             let mut set = BTreeSet::new();
             set.insert(value);
```

Comment:
It looks like this chunk of code could make use of the entry API.

Reply:
Added "investigate this" to my todo list.